src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -67,6 +67,14 @@ using namespace mongo;
 
 /* ****************************************************************************
 *
+* stringFilterP - 
+*/
+extern __thread StringFilter* stringFilterP;
```

Comment:
This is really needed or a leftover of the first version of the PR?

As far as I understand, StringFilter usage is "local":

* In the case of queries it is generated as part of the Scope passed to mongoBackend
* In the case of subscriptions creation/update, it is generated upon csub entry creation (except in the case of not using csub cache, in which case nothing is done regarding StringFilter)
* In the case of updated, StringFilter is retrieved from the cache (in the case of using csub cache) or generated on-the-fly from the q string stored in DB (in the case of not using csub cache).

The situation is similar to fields modeling areas, which are included as part of the Scope in the case of queries. Why in this case (StringFilter) we need a thread var and in that (area stuff) case we don't?